File: Medium/1578. Minimum Time to Make Rope Colorful.java
class Solution {
    public int minCost(String colors, int[] neededTime) {
        char[] colorsToChars = colors.toCharArray();
        int n = colorsToChars.length;
        int sumOfSame = neededTime[0];
        int maxOfSum = neededTime[0];
        int minCost = 0;
        for (int i = 1; i < n; i++) {
            if (colorsToChars[i] == colorsToChars[i - 1]) {
                sumOfSame += neededTime[i];
                maxOfSum = Math.max(neededTime[i], maxOfSum);
            } else {
                minCost += (sumOfSame - maxOfSum);
                sumOfSame = neededTime[i];
                maxOfSum = neededTime[i];
            }
        }
        minCost += (sumOfSame - maxOfSum);
        return minCost;
    }
}

File: Medium/1578. Minimum Time to Make Rope Colorful.py
class Solution:
    def minCost(self, colours: str, times: List[int]) -> int:
        prev_colour = prev_time = None
        result = 0
        for colour, time in zip(colours, times):
            if colour == prev_colour:
                result += min(time, prev_time)
                prev_time = max(prev_time, time)
            else:
                prev_colour = colour
                prev_time = time
        return result